cyfronet-fid/recommender-system
# [#34] add sentry integration

This PR closes #34 issue.

It integrates sentry with `flask` and `celery`. Sentry automatically send events related to unhandled exceptions if `SENTRY_DSN` environmental variable is set in `.env` file.

### How to test:
1. Set `SENTRY_DSN` environmental variable in `.env` as you wish - it can be sentry project dsn that you have access to.
2. Log into chosen project's dashboard.
3. In the local code, add temporarily `print(1/0)` between lines 18 and 19 in [this code here](https://github.com/cyfronet-fid/recommender-system/blob/27ddc03e147910143fb6178e3f99ca73b02a9e68/recommender/api/endpoints/training.py#L18-L19) to trigger flask unhandled exception.
4. In the local code, replace temporarily [this line](https://github.com/cyfronet-fid/recommender-system/blob/27ddc03e147910143fb6178e3f99ca73b02a9e68/recommender/tasks/neural_networks.py#L15) wtih `print(2/0)` to trigger celery unhandled exception.
5. Start local instance of the recommender flask server (test both options: plain flask server and docker deployment).
6. Got to the localhost:5000 (be default) to see recommender API Swagger documentation and use it to trigger request to the training endpoint.

### Expected results
After a while, you should see events related to flask and celery unhandled exceptions in the sentry dashboard (If you do not see anything, wait 15-20 seconds and keep refreshing sentry dashboard).
